gitsync/sync/fetch_pipeline.py

```diff
@@ -50,8 +81,11 @@ async def after_update(self, ref_updates: typ.Dict[str, bytes], respond_to: typ.
             except (ValueError, InterestCanceled, InterestTimeout, InterestNack) as e:
                 logging.warning(f'Fetching error - {type(e)} {e}')
                 continue
-            # TODO: If this is bmeta, fetch refs/head/*
-            pass
+            # If bmeta branch, assume no conflict, set HEAD directly
+            if name.startswith('refs/bmeta/'):
+                self.repo.set_head(name, head)
```

Comment:
By the consistency requirement, you cannot set head of `bmeta` before you actually set the `head` successfully.